k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -138,10 +156,64 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
 
     (_watch(resourceVersion), Closable.ref(close))
   }
+
+  /**
+   * Convert this Watchable into an [[Activity]]
+   * @param resourceVersion whether or not to send the initial resource
+   *                        version of the watched resource. this is a
+   *                        special case due to errors with the
+   *                        ConfigMap interpreter
+   * @param onEvent function called on each [[Watch]] event
+   * TODO: i wish this returned a [[Activity.State]] so that the caller
```

Comment:
probably would want it to return Try.  You never want callers to be able to return Activity.State.Pending here or Bad Things will happen.